bin/ruby-build

```diff
@@ -1081,6 +1081,16 @@ has_broken_mac_openssl() {
   [[ "$RUBY_CONFIGURE_OPTS" != *--with-openssl-dir=* ]]
 }
 
+has_openssl_3() {
+  local openssl_version="$(/usr/bin/openssl version 2>/dev/null || true)"
```

Comment:
What should we do if openssl does not exist/there is system openssl?
I think we should fail in that case and instruct the user to follow https://github.com/rbenv/ruby-build/wiki#suggested-build-environment.
Otherwise we might install an openssl version which would be redundant with the system one, which is bad.

(Regarding detection, in the future, not for this PR, `/usr/bin/openssl` doesn't seem ideal e.g. on macOS where we should probably consider Homebrew and maybe macports. I think we should just pick `openssl` from PATH, so just `openssl version`. But anyway, not for this PR)

Reply:
I decided to warn in such case that there is no system openssl.